staging/src/k8s.io/dynamic-resource-allocation/structured/internal/stable/allocator_stable.go

```diff
@@ -678,6 +678,10 @@ func lookupAttribute(device *draapi.BasicDevice, deviceID DeviceID, attributeNam
 // This allows the logic for subrequests to call allocateOne with the same
 // device index without causing infinite recursion.
 func (alloc *allocator) allocateOne(r deviceIndices, allocateSubRequest bool) (bool, error) {
+	if alloc.ctx.Err() != nil {
+		return false, fmt.Errorf("filter operation aborted: %w", context.Cause(alloc.ctx))
+	}
+
```

Comment:
Some changes like this need to be applied to all forks of the allocator code. Changes for beta features will get applied only to `allocator_incubating.go` (thus not breaking GA!) and changes for alpha features will go into a to-be-added `allocator_experimental.go`.

Reply:
All of the loops end up calling `allocateOne`. I was concerned about checking the context too often, but a performance comparison with and without this didn't show any significant difference.

The most expensive operation, CEL evaluation, checks the context internally.